src/definitions.ts

```diff
@@ -163,10 +164,24 @@ export interface UserDependencies {
   service: ISingleUserService;
 }
 
+export interface UserSocial {
+  [provider: string]: {
```

Comment:
Let's make a `UserSocialProvider` interface and explicitly define the networks, reusing the interface for each network.